tests/src/PatternAssertion/DescriptionListAssert.php

```diff
@@ -0,0 +1,202 @@
+    return [
+      'items' => [
+        [$this, 'assertItems'],
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function assertBaseElements(string $html, string $variant): void {
+    $crawler = new Crawler($html);
+    $description_list = $crawler->filter('body > .bcl-description-list');
+    self::assertCount(1, $description_list);
+  }
+
+  /**
+   * Asserts the items of the pattern.
+   *
+   * @param array $expected_items
+   *   The expected item values.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The DomCrawler where to check the element.
+   */
+  protected function assertItems(array $expected_items, Crawler $crawler): void {
+    $labels = $this->extractLabels($expected_items, 'term');
+    $values = $this->extractLabels($expected_items, 'definition');
+    $dt_icons = $this->extractIcons($expected_items, 'term');
+    $dd_icons = $this->extractIcons($expected_items, 'definition');
+
+    $this->assertLabels($labels, $crawler);
+    $this->assertIcons($dt_icons, 'dt', $crawler);
+    $this->assertIcons($dd_icons, 'dd', $crawler);
+    $this->assertValues($values, $crawler);
+  }
+
+  /**
+   * Extracts labels from expected items.
+   *
+   * @param array $expected_items
+   *   The expected item values.
+   * @param string $key
+   *   The key to extract labels for ('term' or 'definition').
+   *
+   * @return array
+   *   The extracted labels.
+   */
+  private function extractLabels(array $expected_items, string $key): array {
+    $labels = [];
+    foreach ($expected_items as $item) {
+      if (isset($item[$key])) {
+        $labels = array_merge($labels, $this->extractLabel($item[$key]));
+      }
+    }
+    return $labels;
+  }
+
+  /**
+   * Extracts label from an item.
+   *
+   * @param mixed $item
+   *   The item to extract label from.
+   *
+   * @return array
+   *   The extracted labels.
+   */
+  private function extractLabel($item): array {
+    $labels = [];
+    if ($item === NULL) {
+      return $labels;
+    }
+
+    if (is_array($item)) {
+      foreach ($item as $subItem) {
+        if (isset($subItem['label'])) {
+          $labels[] = is_array($subItem['label']) ? strip_tags($subItem['label'][0]['#markup'] ?? $subItem['label'][0]) : strip_tags($subItem['label']);
+        }
+      }
+    }
+    elseif (is_string($item)) {
+      $labels[] = strip_tags($item);
+    }
+    return $labels;
+  }
+
+  /**
+   * Extracts icons from expected items.
+   *
+   * @param array $expected_items
+   *   The expected item values.
+   * @param string $key
+   *   The key to extract icons for ('term' or 'definition').
+   *
+   * @return array
+   *   The extracted icons.
+   */
+  private function extractIcons(array $expected_items, string $key): array {
+    $icons = [];
+    foreach ($expected_items as $item) {
+      if (isset($item[$key])) {
+        $icons = array_merge($icons, $this->extractIcon($item[$key]));
```

Comment:
same comments as for the other method pairs

Reply:
done.